src/t.jl

```diff
@@ -28,18 +28,18 @@ export OneSampleTTest, TwoSampleTTest, EqualVarianceTTest,
 @compat abstract type TTest <: HypothesisTest end
 @compat abstract type TwoSampleTTest <: TTest end
 
-pvalue(x::TTest; tail=:both) = pvalue(TDist(x.df), x.t; tail=tail)
-
-default_tail(test::TTest) = :both
+pvalue(x::TTest; tail=x.tail) = pvalue(TDist(x.df), x.t; tail=tail)
+tail(x::TTest)  = x.tail    # defaults set by constructors
```

Comment:
> Adding a tail field to all types, which is not used by several of them, is not a quite clean solution. I would rather add the field to those types that actually can make use of it. These are quite many, though: Below, you find a list of types for which the pvalue function has a tail parameter, and those types for which pvalue has a fixed tail.
> 
> I will add a tail field to those types, for which the pvalue function takes a tail parameter.

Yes, of course, no need to add the field to tests which don't need it, overloading the method will be enough for these.

> The alpha value is relevant to all tests. Therefore, I will add an alpha field to all types.

Right.

> I will also add a simple function obtaining the test outcome:
> 
> reject_h0(test::HypothesisTest) = pvalue(test) <= test.alpha

I'd rather not add this function, or at least not in this PR, as that's a different issue which deserves some discussion.